MacOS Node artifact build broken on master

Failing both on Jenkins and Kokoro with the same error
https://sponge.corp.google.com/target?id=b0ce932f-3a58-4bd0-a1db-2f1bf16b7d69&target=github/grpc&searchFor=&show=ALL&sortBy=STATUS

https://grpc-testing.appspot.com/view/Artifacts/job/gRPC_build_artifacts/491/#showFailuresLink

```
CXX(target) Release/obj.target/grpc_node/src/node/ext/byte_buffer.o
  CXX(target) Release/obj.target/grpc_node/src/node/ext/call.o
In file included from ../src/node/ext/call.cc:26:
../src/node/ext/call.h:35:12: error: no member named 'unique_ptr' in namespace 'std'
using std::unique_ptr;
      ~~~~~^
../src/node/ext/call.h:36:12: error: no member named 'shared_ptr' in namespace 'std'
using std::shared_ptr;
      ~~~~~^
../src/node/ext/call.h:100:21: error: use of undeclared identifier 'unique_ptr'
typedef std::vector<unique_ptr<Op>> OpVec;
                    ^
../src/node/ext/call.h:100:32: error: 'Op' does not refer to a value
typedef std::vector<unique_ptr<Op>> OpVec;
                               ^
../src/node/ext/call.h:87:7: note: declared here
class Op {
      ^
../src/node/ext/call.h:102:32: error: unknown type name 'OpVec'
  tag(Nan::Callback *callback, OpVec *ops, Call *call,
                               ^
../src/node/ext/call.h:106:3: error: unknown type name 'OpVec'
  OpVec *ops;
  ^
../src/node/ext/call.cc:38:12: error: no member named 'unique_ptr' in namespace 'std'
using std::unique_ptr;
      ~~~~~^
../src/node/ext/call.cc:39:12: error: no member named 'shared_ptr' in namespace 'std'
using std::shared_ptr;
      ~~~~~^
../src/node/ext/call.cc:442:30: error: unknown type name 'OpVec'
tag::tag(Callback *callback, OpVec *ops, Call *call, Local<Value> call_value)
                             ^
../src/node/ext/call.cc:459:17: error: use of undeclared identifier 'unique_ptr'
    for (vector<unique_ptr<Op> >::iterator it = tag_struct->ops->begin();
                ^
../src/node/ext/call.cc:459:28: error: 'Op' does not refer to a value
    for (vector<unique_ptr<Op> >::iterator it = tag_struct->ops->begin();
                           ^
../src/node/ext/call.h:87:7: note: declared here
class Op {
      ^
../src/node/ext/call.cc:459:32: error: expected unqualified-id
    for (vector<unique_ptr<Op> >::iterator it = tag_struct->ops->begin();
                               ^
../src/node/ext/call.cc:460:10: error: expected ';' in 'for' statement specifier
         it != tag_struct->ops->end(); ++it) {
         ^
../src/node/ext/call.cc:460:10: error: use of undeclared identifier 'it'
../src/node/ext/call.cc:460:42: error: use of undeclared identifier 'it'
         it != tag_struct->ops->end(); ++it) {
                                         ^
../src/node/ext/call.cc:461:20: error: use of undeclared identifier 'it'
      Op *op_ptr = it->get();
                   ^
../src/node/ext/call.cc:472:15: error: use of undeclared identifier 'unique_ptr'
  for (vector<unique_ptr<Op> >::iterator it = tag_struct->ops->begin();
              ^
../src/node/ext/call.cc:472:26: error: 'Op' does not refer to a value
  for (vector<unique_ptr<Op> >::iterator it = tag_struct->ops->begin();
                         ^
../src/node/ext/call.h:87:7: note: declared here
class Op {
      ^
../src/node/ext/call.cc:472:30: error: expected unqualified-id
  for (vector<unique_ptr<Op> >::iterator it = tag_struct->ops->begin();
                             ^
fatal error: too many errors emitted, stopping now [-ferror-limit=]
20 errors generated.
make: *** [Release/obj.target/grpc_node/src/node/ext/call.o] Error 1
```

--- a/binding.gyp
+++ b/binding.gyp
@@ -175,21 +175,27 @@
       }],
       ['OS == "mac"', {
         'xcode_settings': {
-          'MACOSX_DEPLOYMENT_TARGET': '10.9'
+          'OTHER_CFLAGS': [
+              '-g',
+              '-Wall',
+              '-Wextra',
+              '-Werror',
+              '-Wno-long-long',
+              '-Wno-unused-parameter',
+              '-DOSATOMIC_USE_INLINED=1',
+          ],
+          'OTHER_CPLUSPLUSFLAGS': [
+              '-g',
+              '-Wall',
+              '-Wextra',
+              '-Werror',
+              '-Wno-long-long',
+              '-Wno-unused-parameter',
+              '-DOSATOMIC_USE_INLINED=1',
+            '-stdlib=libc++',
+            '-std=c++11'
+          ],
         },
-        'OTHER_CFLAGS': [
-            '-g',
-            '-Wall',
-            '-Wextra',
-            '-Werror',
-            '-Wno-long-long',
-            '-Wno-unused-parameter',
-            '-DOSATOMIC_USE_INLINED=1',
-        ],
-        'OTHER_CPLUSPLUSFLAGS': [
-          '-stdlib=libc++',
-          '-std=c++11'
-        ],
       }]
     ]
   },
@@ -508,6 +514,13 @@
             'third_party/boringssl/ssl/tls_method.c',
             'third_party/boringssl/ssl/tls_record.c',
           ],
+          'conditions': [
+            ['OS == "mac"', {
+              'xcode_settings': {
+                'MACOSX_DEPLOYMENT_TARGET': '10.9'
+              }
+            }]
+          ]
         },
       ],
     }],
@@ -626,6 +639,13 @@
         'src/core/lib/support/tmpfile_windows.c',
         'src/core/lib/support/wrap_memcpy.c',
       ],
+      'conditions': [
+        ['OS == "mac"', {
+          'xcode_settings': {
+            'MACOSX_DEPLOYMENT_TARGET': '10.9'
+          }
+        }]
+      ]
     },
     {
       'target_name': 'grpc',
@@ -889,6 +909,13 @@
         'src/core/ext/filters/workarounds/workaround_utils.c',
         'src/core/plugin_registry/grpc_plugin_registry.c',
       ],
+      'conditions': [
+        ['OS == "mac"', {
+          'xcode_settings': {
+            'MACOSX_DEPLOYMENT_TARGET': '10.9'
+          }
+        }]
+      ]
     },
     {
       'include_dirs': [
@@ -914,6 +941,11 @@
           'ldflags': [
             '-Wl,-wrap,memcpy'
           ]
+        }],
+        ['OS == "mac"', {
+          'xcode_settings': {
+            'MACOSX_DEPLOYMENT_TARGET': '10.9'
+          }
         }]
       ],
       "target_name": "grpc_node",

--- a/binding.gyp
+++ b/binding.gyp
@@ -193,7 +193,8 @@
               '-Wno-unused-parameter',
               '-DOSATOMIC_USE_INLINED=1',
             '-stdlib=libc++',
-            '-std=c++11'
+            '-std=c++11',
+            '-Wno-error=deprecated-declarations'
           ],
         },
       }]

--- a/templates/binding.gyp.template
+++ b/templates/binding.gyp.template
@@ -176,7 +176,8 @@
                 '${item}',
               % endfor
               '-stdlib=libc++',
-              '-std=c++11'
+              '-std=c++11',
+              '-Wno-error=deprecated-declarations'
             ],
             % endif
           },

--- a/templates/binding.gyp.template
+++ b/templates/binding.gyp.template
@@ -165,19 +165,21 @@
         }],
         ['OS == "mac"', {
           'xcode_settings': {
-            'MACOSX_DEPLOYMENT_TARGET': '10.9'
+            % if defaults['global'].get('CPPFLAGS', None) is not None:
+            'OTHER_CFLAGS': [
+              % for item in defaults['global'].get('CPPFLAGS').split():
+                '${item}',
+              % endfor
+            ],
+            'OTHER_CPLUSPLUSFLAGS': [
+              % for item in defaults['global'].get('CPPFLAGS').split():
+                '${item}',
+              % endfor
+              '-stdlib=libc++',
+              '-std=c++11'
+            ],
+            % endif
           },
-          % if defaults['global'].get('CPPFLAGS', None) is not None:
-          'OTHER_CFLAGS': [
-            % for item in defaults['global'].get('CPPFLAGS').split():
-              '${item}',
-            % endfor
-          ],
-          'OTHER_CPLUSPLUSFLAGS': [
-            '-stdlib=libc++',
-            '-std=c++11'
-          ],
-          % endif
         }]
       ]
     },
@@ -201,6 +203,13 @@
               '${source}',
               % endfor
             ],
+            'conditions': [
+              ['OS == "mac"', {
+                'xcode_settings': {
+                  'MACOSX_DEPLOYMENT_TARGET': '10.9'
+                }
+              }]
+            ]
           },
           % endif
           % endfor
@@ -282,6 +291,13 @@
           '${source}',
           % endfor
         ],
+        'conditions': [
+          ['OS == "mac"', {
+            'xcode_settings': {
+              'MACOSX_DEPLOYMENT_TARGET': '10.9'
+            }
+          }]
+        ]
       },
       % endif
       % endfor
@@ -317,6 +333,11 @@
             'ldflags': [
               '-Wl,-wrap,memcpy'
             ]
+          }],
+          ['OS == "mac"', {
+            'xcode_settings': {
+              'MACOSX_DEPLOYMENT_TARGET': '10.9'
+            }
           }]
         ],
         "target_name": "${module.name}",

